docs/grpc_disruption/interceptor_algorithm.md

```diff
@@ -0,0 +1,169 @@
+# gRPC Disruption Inceptor
+
+When the interceptor recognizes a query's endpoint as one which is actively getting disrupted, the interceptor generates a random integer from `0` to `100`, and consults an array caching the `PercentToAlteration` mapping. Provided the random integer, this hasmap identifies the disruption to apply to a query. The mapping represents user preference the proportion of queries affected by each alteration. The user cannot sepcify queryPercents for a single endpoint which sum to over `100%`. You can see an example below of an array that does not define all 100% of possible requests. You can see the next section for an example of a mapping which only has 25 entires.
```

Comment:
Is this still true? I notice this is alternating between mentioning hashmap vs array

Reply:
oof, yeah. I will call it a hashmap because that's more true to the design, and mention it's an array underneath the hood.